app/services/preserved_object_handler.rb

```diff
@@ -174,19 +176,25 @@ def update_version
   private
 
   def moab_validation_errors
-    object_dir = "#{storage_location}/#{DruidTools::Druid.new(druid).tree.join('/')}"
-    moab = Moab::StorageObject.new(druid, object_dir)
-    object_validator = Stanford::StorageObjectValidator.new(moab)
-    moab_errors = object_validator.validation_errors(Settings.moab.allow_content_subdirs)
-    @ran_moab_validation = true
-    if moab_errors.any?
-      moab_error_msgs = []
-      moab_errors.each do |error_hash|
-        error_hash.each_value { |msg| moab_error_msgs << msg }
+    # validation is a bit expensive, so memoize the result.  keeps us from having to worry about a
+    # performance hit if a caller wants to check whether there are validation errors from more than
+    # one conditional statement.
```

Comment:
i think you can leave out this comment - memoization is a common practice and devs know why it is used.